File: src/main/java/ru/citeck/idea/utils/CiteckPsiUtils.java
package ru.citeck.idea.utils;

import com.intellij.json.psi.JsonObject;
import com.intellij.json.psi.JsonProperty;
import com.intellij.json.psi.JsonPsiUtil;
import com.intellij.json.psi.JsonValue;
import com.intellij.openapi.editor.Document;
import com.intellij.psi.PsiDocumentManager;
import com.intellij.psi.PsiFile;
import com.intellij.psi.xml.XmlFile;
import com.intellij.psi.xml.XmlTag;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.nio.charset.StandardCharsets;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

public class CiteckPsiUtils {

    public static XmlTag getRootTag(@NotNull PsiFile psiFile) {
        if (!(psiFile instanceof XmlFile)) {
            return null;
        }
        return ((XmlFile) psiFile).getRootTag();
    }

    public static Map<String, String> getProperties(@NotNull JsonObject jsonObject, String... property) {
        Set<String> properties = Set.of(property);
        return jsonObject
            .getPropertyList()
            .stream()
            .filter(jsonProperty -> properties.contains(jsonProperty.getName()))
            .collect(Collectors.toMap(JsonProperty::getName, CiteckPsiUtils::getValue));
    }

    public static String getProperty(@NotNull JsonObject jsonObject, String property) {
        return getValue(jsonObject.findProperty(property));
    }

    public static @NotNull String getValue(@Nullable JsonProperty jsonProperty) {
        if (jsonProperty == null) {
            return "";
        }
        JsonValue jsonValue = jsonProperty.getValue();
        if (jsonValue == null) {
            return "";
        }
        String value = jsonValue.getText();
        if (value == null) {
            return "";
        }
        return JsonPsiUtil.stripQuotes(value);
    }

    public static void setContent(PsiFile psiFile, String content) throws Exception {
        Document document = PsiDocumentManager.getInstance(psiFile.getProject()).getDocument(psiFile);
        if (document != null) {
            document.setText(content);
            return;
        }
        psiFile.getVirtualFile().setBinaryContent(content.getBytes(StandardCharsets.UTF_8));
    }
}

File: src/main/kotlin/ru/citeck/idea/utils/CiteckPsiUtils.kt
package ru.citeck.idea.utils

import com.intellij.json.psi.JsonObject
import com.intellij.json.psi.JsonProperty
import com.intellij.json.psi.JsonPsiUtil
import com.intellij.psi.PsiDocumentManager
import com.intellij.psi.PsiFile
import com.intellij.psi.xml.XmlFile
import com.intellij.psi.xml.XmlTag
import java.nio.charset.StandardCharsets

object CiteckPsiUtils {

    fun getRootTag(psiFile: PsiFile): XmlTag? {
        if (psiFile !is XmlFile) {
            return null
        }
        return psiFile.rootTag
    }

    fun getProperties(jsonObject: JsonObject, vararg property: String): Map<String, String> {
        val properties = property.toSet()
        return jsonObject.propertyList
            .filter { properties.contains(it.name) }
            .associate { it.name to getValue(it) }
    }

    fun getProperty(jsonObject: JsonObject, property: String): String {
        return getValue(jsonObject.findProperty(property))
    }

    fun getValue(jsonProperty: JsonProperty?): String {
        if (jsonProperty == null) {
            return ""
        }
        val jsonValue = jsonProperty.value ?: return ""
        val value = jsonValue.text ?: return ""
        return JsonPsiUtil.stripQuotes(value)
    }

    fun setContent(psiFile: PsiFile, content: String) {
        val document = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile)
        if (document != null) {
            document.setText(content)
            return
        }
        psiFile.virtualFile.setBinaryContent(content.toByteArray(StandardCharsets.UTF_8))
    }
}
